## Capítulo 1.1.: Modelos estadísticos para control de calidad

### Caso 1.1.1: Análisis de capacidad de proceso

In [20]:
import numpy as np
import math
import scipy.stats as st

#### Ejemplo 4:

Un proceso está bajo control con $\bar{X'} = 100$ y $s = 1.05$. Las especificaciones del proceso son $95 \pm 10$.La característica de calidad tiene una distribución normal.
* **a) Estimar la Capacidad Potencial (Cp).**
* **b) Estimar la Capacidad Real (Cpk).**
* **c) ¿Qué sucedería con la fluctuación natural del proceso si éste se corrigiera para operar en la especificación nominal?**

**a)**

R/ Se sabe que el índice de capacidad del proceso Cp esta dado por:

$$\rightarrow C_{p}=\frac{T}{6\sigma'}$$

Y:
$$\rightarrow T = 2\nabla = LSE - LIE$$

Donde:
* $\nabla$ es la discrepancia y de ahora en adelante se le asignará la letra 'D', 
* $\sigma'$ que es la desviación estándar se le asignará la letra 's',y 
* 'T' es la tolerancia. 

A continuación se procederá determinar la capacidad del proceso:

In [21]:
D = 10
LIE = 95 - D
LSE = 95 + D
T = 2*D
s = 1.05 # Se toma s como el mejor estimador de sigma.

Cp = T/(6*s)
print(Cp)

3.1746031746031744


Dado que $C_{p} \geq 2.0$ entonces esto por lo pronto indica que el proceso tiene capacidad de fabricar producto dentro de las especificaciones, y según los criterios de evaluación indica que se trata de un proceso 6 sigma de clase mundial.

**b)** 

R/ Cuando el proceso no está centrado, se emplea el **índice de capacidad unilateral** o **índice de capacidad real (Cpk)**:

$$\rightarrow C_{pk}=min(C_{pki}, C_{pks})$$

$$\rightarrow C_{pk}=min(C_{pki}=\frac{\bar{X'}-LIE}{3\sigma'}, C_{pks}=\frac{LSE-\bar{X'}}{3\sigma'})$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación del proceso.

Por tanto, el índice Cpk es:

In [22]:
X = 100 # Se toma X' como el mejor estimador de mu

Cpki = np.round((X - LIE)/(3*s),2)
Cpks = np.round((LSE - X)/(3*s),2)

Cpk = np.min([Cpki, Cpks])

print("Cpki = "+ str(Cpki))
print("Cpks = "+ str(Cpks))
print("Cpk = "+ str(Cpk))

Cpki = 4.76
Cpks = 1.59
Cpk = 1.59


Cpk $\approx$ 1.59, esto indica que el proceso es realmente capaz y tiende a cumplir con ambas especificaciones de calidad.

**c)**

R/ Para obtener la porción caída del proceso, se tiene que calcular los límites de fluctuación natural del proceso, dado $\bar{X'} = VN$, es decir $\bar{X'}= 95$. Los límites de fluctiación natural del proceso están dados por la siguiente fórmula:

$$\rightarrow LFN_{i,s} = \bar{X'} \pm 3\sigma'$$

Donde:

* $\bar{X'}$ es la media del proceso, y
* $\sigma'$ es la desviación estándar del proceso.

Entonces, los límites de fluctuación son:

In [23]:
X = 95

LIFN = np.round(X - 3*s,4)
LSFN = np.round(X + 3*s,4)

print("LIFN = " + str(LIFN) + " y LSFN = " + str(LSFN))

LIFN = 91.85 y LSFN = 98.15


Se compara si, dada la nueva media ($\bar{X'} = 95$), LIFN es mayor o igual a LIE y si LSFN es menor o igual a LSE:

In [19]:
print(LIFN>=LIE, LSFN<=LSE)

True True


Se tiene que sin importar si la media del proceso se corrigiera para operar en la especificación nominal el proceso seguiría dentro de los límites de especificación inferior y superior.